File: autotest_master/api/company.py
#-*- coding: UTF-8 -*-
#import http.client ,urllib
import urllib
import httplib
import unittest
import json
import requests

HOST = "rongtest4.36kr.com"
HIGH_INVESTOR = {"Content-type":"application/x-www-form-urlencoded",'Cookie':'kr_stat_uuid=fxaPr23881085; kr_plus_id=69336; kr_plus_token=067a203b3e4f80a8dd81c2f9384052b096248fd5; _gat=1; _ga=GA1.2.370057440.1432865120; krid_user_version=6; krid_user_id=123897; _auth_session=c891f612700e53c68f9904fbab5dec71; _krypton_session=2922adfe93d42a942166ed832e1deeed'}


class TestCompany(unittest.TestCase):

    def setUp(self):
        print ("start test")

    # def raises_(expected_exc, fun, *args, **kwargs):
    #     try:
    #         fun(*args, **kwargs)
    #     except Exception as exc:
    #         assert type(exc) == expected_exc
    #         return
    #     raise AssertionError("Exception %s not raised" % expected_exc.__name__)

    def send(self, methed, api, headers=None, data=None):
        if not headers:
            headers = {}
        if not data:
            data = ""
        else:
            data = urllib.urlencode(data)
            print (data)

        conn = httplib.HTTPConnection(HOST)
        conn.request(methed, api, data, headers)
        response = conn.getresponse()
        code = response.status
        data = response.read()
        result = json.loads(data)
        print result
        code = result['code']
        print (code)
        return code

    def get_send(self, api, headers=None, params=None):
        if not headers:
            headers = {}
        conn = requests.get(api,params=params, headers = headers)

    def test_company_list(self):
        #methed = "GET"
        api = "%s/api/company?fincestatus=2&page=1&type=" % HOST
        params = {'fincestatus':2}
        self.assertEqual(self.send(methed,api), 0)

    def test_count_all(self):
        methed = "GET"
        api = "/api/company/count-all"
        headers= HIGH_INVESTOR
        self.assertEqual(self.send(api, headers), 0)

    def test_count_common(self):
        methed = "GET"
        api = "/api/company/count-common"
        self.assertEqual(self.send(methed, api), 0)

    def test_get_company(self):
        methed = "GET"
        api = "/api/company/142910"
        headers= HIGH_INVESTOR
        self.assertEqual(self.send(methed, api, headers), 0)

    def test_check_company(self):
        methed = "GET"
        api    = "/api/suggest/company?wd=ceshiceshi"
        headers= HIGH_INVESTOR
        self.assertEqual(self.send(methed, api, headers), 0)

    def test_update_company(self):
        methed = "PUT"
        api    = "/api/company/142900"
        headers= HIGH_INVESTOR
        data = {"cid" : 142900,
                "website" : "http://www.baidu.com",
                "brief":'第一次修改', "industry":"AUTO",
                "operationStatus":"OPEN"}
        self.assertEqual(self.send(methed,api,headers,data),0)

    def test_set_managed(self):
        methed = "PUT"
        api    = "/api/company/142900"
        headers= HIGH_INVESTOR
        data = {"manager":"68717"}
        self.assertEqual(self.send(methed,api,headers,data),0)

    def test_create_company(self):
        methed = "POST"
        api    = "/api/company"
        headers= HIGH_INVESTOR
        data   = {"name":"测试接口test1", "brief":"测试接口demo","type":"FOUNDER","position":"CEO","startDate":"2013-05-01 12:00:00","endDate":""}
        self.assertEqual(self.send(methed,api,headers,data),0)

    def tearDown(sefl):
        print ("     \ntest_done")

def suite():
    suite = unittest.TestSuite()
   # suite.addTest(TestCompany("test_company_list"))
    suite.addTest(TestCompany("test_count_all"))
    return suite
# if __name__ == '__main__':
#     unittest.main()
if __name__ == '__main__':
    unittest.main(defaultTest = 'suite')

File: autotest_master/api/company.rb
require 'rubygems'
require 'json'
require 'net/http'
require 'uri'
require 'erb'
require File.dirname(__FILE__)+'/../global'

module KR
  module API
    extend self
          attr_accessor :context
          def self.set(context)
          end
          def self.json_parse(res_body)
              puts res_body
              result = JSON.parse(res_body)
              result.each{|args|
              puts args
             }
              return check_result(result)
          end
          def self.get_response(uri, cookie=nil, parame=nil)
            if parame != nil
               uri.query = URI.encode_www_form(parame)
            end
              req = Net::HTTP::Get.new(uri)
            if cookie != nil
              req['Cookie'] = cookie
            end
              res = Net::HTTP.start(uri.hostname, uri.port) {|http| http.request(req)}
              result = json_parse(res.body)
              return result
          end
          def self.post_response(uri, cookie, parame=nil)
              req = Net::HTTP::Post.new(uri)
              req.set_form_data(parame)
              req['Cookie'] = cookie
              res = Net::HTTP.start(uri.hostname, uri.port) {|http| http.request(req)}
              result = json_parse(res.body)
              return result
          end
          def self.put_response(uri,cookie,parame=nil)
              req = Net::HTTP::Put.new(uri)
              if parame != nil
                  req.set_form_data(parame)
              end
              req['Cookie'] = cookie
              res = Net::HTTP.start(uri.hostname, uri.port) {|http| http.request(req)}
              result = json_parse(res.body)
              return result
          end
          def check_result(result)
              r = ""
              if result.has_key? 'Error'
                  r += "failed to web service error"
               end
               if result["code"] == 0
                  r += "succeed to test the company?#{}" +"\s" + result["code"].to_s
               else
                  r += "failed to test the company?type=managed" +"\s" + result["code"].to_s
               end
               puts r
              return r
          end


      class COMPANY
          def company_list(args)#########根据条件获取公司列表###########
              cookie = args[:cookie]
              uri = URI('http://rong.test.36kr.com/company/')

              parame = {"fincestatus" => args[:fincestatus], 'type' => args[:type], 'page' => args[:page]}
              result = API.get_response(uri, cookie, parame)
          end
          def count_all(args)###获取全部融资公司个数（对优质投资人可见的和不可见的)##############
              r = ""
              cookie = args[:cookie]
              uri = URI('http://rong.test.36kr.com/company/count-all')
              result = API.get_response(uri, cookie)
          end
          def count_common(args)#########获取普通融资公司个数（对非优质投资人可见的)###########
              r = ""
              cookie = args[:cookie]

              uri = URI("http://rong.test.36kr.com/company/count-common")
              result = API.get_response(uri, cookie)
          end
          def get_company(args)########获取公司的基本信息，介绍，链接，融资信息，标签###########
              r = ""
              cookie = args[:cookie]
              cid    = args[:cid]

              uri = URI("http://rong.test.36kr.com/company/#{cid}")
              result = API.get_response(uri, cookie)
          end
          def check_company(args)#######根据名称检查公司是否可创建，如果已有同名的公司，则返回它的数据##########
              r = ""
              cookie = args[:cookie]
              parame = {"action" => "checkName", "name" => args[:name]}

              uri = URI("http://rong.test.36kr.com/company")
              result = API.get_response(uri, cookie, parame)
          end
          def get_managed(args)###获取用户管理的公司########
              r = ""
              cookie = args[:cookie]
              uri = URI('http://rong.test.36kr.com/company?type=managed')
              result = API.get_response(uri, cookie)
          end
          def create_fast(args)######快速创建公司########
              r = ""
              cookie = args[:cookie]
              parame = {'mode' => args[:mode] , 'name' => args[:name] , 'brief' => args[:brief], :companySource => args[:companySource]}
              puts parame
              uri = URI("http://rong.test.36kr.com/company")
              result = API.post_response(uri, cookie, parame)
          end
          def update_company(args)######更新公司信息########
              r = ""
              cookie = args[:cookie]
              cid    = args[:cid]
              parame = {'website' => args[:website], 'brief' => args[:brief], 'industry' => args[:industry], 'startDate' => args[:startDate], 'operationStatus' => args[:operationStatus]}
              uri = URI("http://rong.test.36kr.com/company/#{cid}")
              result = API.put_response(uri, cookie, parame)
          end
          def set_managed(args)######设置管理员#########
              r = ""
              cookie = args[:cookie]
              cid    = args[:cid]
              parame = {'manager' => args[:manager]}
              uri = URI("http://rong.test.36kr.com/company/#{cid}")
              result = API.put_response(uri, cookie, parame)
          end
          def claim_company(args)#####认领公司#########
              r = ""
              cookie = args[:cookie]
              cid    = args[:cid]
              parame = {'type' => args[:type]}
              uri = URI("http://rong.test.36kr.com/company/#{cid}?action=claim")
              result = API.put_response(uri, cookie, parame)

          end
          def create_company(args)######正常创建公司－进入草稿模式###########
              r = ""
              cookie = args[:cookie]
              parame = {'name' => args[:name], 'brief' => args[:brief], 'type' => args[:type], 'position' => args[:position], 'startDate' => args[:startDate], 'endDate' => args[:endDate]}
              uri = URI("http://rong.test.36kr.com/company/")
              result = API.post_response(uri, cookie, parame)
          end
          def post_user
              parame = {:tpye => "email", :email => "zhuzhutest111@163.com", "name" => "测试"}
              uri = URI("http://rong.dev.36kr.com/api/user/invite")
              result = API.post_response(uri, parame)
          end
          # diao = COMPANY.new
          # diao.create_company

      end
  end
end